### PR TITLE
WIP:add npm-audit and snyk to ci

### PR DESCRIPTION
**Pull request by yulya9060**

### CODE REVIEW

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-03c20129`

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-3f3414a4`

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-78419362`

File: .travis.yml [hunk 1]
```diff
@@ -46,3 +46,14 @@ deploy:
...
       tags: false
+
+after_script:
+  - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo
+    $TRAVIS_PULL_REQUEST_BRANCH; fi)
+  - echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, PR=$PR, BRANCH=$BRANCHgi"
+    # the command terminates with an error code, even if it does not find vulnerabilities of the specified level.
+    # For the behavior we need, we set a condition under which the command terminates with an error code
+    # only if there is a vulnerability of this level in its output
+  - bash -c 'output=`yarn audit --level=high`; ret_code=$?; echo "$output"; [[ $output =~ "high" ]]
```
[the remark this answers is not in the document]
Comment (yulya9060) [reply .travis.yml]: да, поправила

File: .travis.yml [hunk 2]
```diff
@@ -46,3 +46,14 @@ deploy:
     on:
       branch: master
       tags: false
+
+after_script:
+  - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo
```
[the remark this answers is not in the document]
Comment (yulya9060) [reply .travis.yml]: ок, убрала, теперь проверка будет идти по значению $TRAVIS_PULL_REQUEST_BRANCH

File: package.json [hunk 3]
```diff
@@ -20,7 +20,8 @@
     "watch": "yarn build --watch",
     "//": "---        NPM hooks section        ---",
     "postinstall": "move-bin; true",
-    "prepublishOnly": "yarn build"
+    "prepublishOnly": "yarn build",
+    "snyk": "snyk test --dev --severity-threshold=medium"
```
[the remark this answers is not in the document]
Comment (yulya9060) [reply package.json]: поправила

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-049ff86d`

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-8d699a46`

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yulya9060) [reply .travis.yml]: TRAVIS_BRANCH:
- for push builds, or builds not triggered by a pull request, this is the name of the branch.
- **for builds triggered by a pull request this is the name of the branch targeted by the pull request.**
- for builds triggered by a tag, this is the same as the name of the tag (TRAVIS_TAG).

Дима подсказал решение, что если это PR, то просто выходить с exit code 0, а если нет, то проверять по значению TRAVIS_BRANCH

File: .travis.yml [hunk 4]
```diff
@@ -46,3 +46,14 @@ deploy:
...
+    $TRAVIS_PULL_REQUEST_BRANCH; fi)
+  - echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, PR=$PR, BRANCH=$BRANCHgi"
```
[the remark this answers is not in the document]
Comment (yulya9060) [reply .travis.yml]: убрала)

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-8b843ce0`

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-44fae8a2`

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-2cdbe3c0`

**Comment by nezed** [comment]
New version of package: `4.5.6-MDT-596-check-project-for-vulnerabilities-1c9f7140`

**Pull request merged**